basics_of_logging_in_sst-core/Example01/ExampleComponent.cc

```diff
@@ -0,0 +1,97 @@
+// Created for SST-Core Version (9.1.0)
+//
+#include "ExampleComponent.h"
+#include <iostream>
+
+using namespace Example01;
+
+// Component constructor.
+//
+ExampleComponent::ExampleComponent(SST::ComponentId_t id, SST::Params &params) :
+    SST::Component(id),
+    componentId_(id),
+    clockTickCount_(0)
+{
+    // Read in the parameters from the python config file.  See SST_ELI_DOCUMENT_PARAMS
+    // for an explanation of what each parameter represents.
+    //
+    std::string clock = 
+        params.find<std::string>("clock", "1GHz");
+    clockTicks_ = static_cast<uint64_t>(
+        params.find<int>("clockTicks", 10));
+
+    // Create the logger.
+    //
+    logger_ = SST::Output("Time=@t; File=@f; Func=@p; Line=@l; Thread=@I -- ", LogLevel::Trace, 0x00, SST::Output::STDOUT);
+    logger_.verbose(CALL_INFO, LogLevel::Trace, 0x00, "Entering constructor for component id %lu\n", componentId_);
```

Comment:
Output::verbose() calls have a non-zero overhead and adding so many calls can greatly slow simulation.  For anything that is used primarily for debug (which in this example is probably the DEBUG and TRACE levels should call Output::debug(), which will be compiled out unless debug is enabled at configure time.  Users should not be encouraged to use verbose for debug messages.